troposphere/static/js/components/modals/instance/InstanceResumeModal.jsx

```diff
@@ -25,38 +28,72 @@ export default React.createClass({
     // ------
     //
 
-    renderBody: function() {
+    renderBody: function(multiple, instances) {
+        let operativeNoun = multiple ? "these instances": "this instance",
+            instanceList = (<div/>);
+
+        if (instances) {
+            instanceList = (
+            <div>
+                <ul>
+                {instances.map((i) => {
+                    return (
+                    <li key={i.get("uuid")}>
+                        <strong>
+                            {`${i.get("name")}`}
+                        </strong>
+                        <ul>
+                            <li style={{listStyleType: "square"}}>
+                                {`started on: ${i.get("start_date")}`}
+                            </li>
+                        </ul>
+                    </li>
+                    );
+                 })}
+                </ul>
+            </div>
+            );
+        }
+
         return (
         <div>
+            {instanceList}
             <p>
-                {"Would you like to resume this instance?"}
+                {`Would you like to resume ${operativeNoun}?`}
             </p>
             <p>
-                Your instance's IP address may change once it resumes.
+                An instance's IP address may change when it resumes.
             </p>
         </div>
         );
     },
 
     render: function() {
+        // AUTHOR'S NOTE - not sure if I should extract this filter predicate into
+        // a utility function to be shared across the <Instance*Modal/>s ...
+        let { resources } = this.props,
+            instances = resources ? resources.filter((r) => r instanceof Instance) : null,
+            multi = instances && instances.length > 1,
+            titleNoun = multi ? "Instances" : "Instance",
+            buttonText = multi ? "these instances" : "this instance";
 
         return (
         <div className="modal fade">
             <div className="modal-dialog">
                 <div className="modal-content">
                     <div className="modal-header">
                         {this.renderCloseButton()}
-                        <h1 className="t-title">Resume Instance</h1>
+                        <h1 className="t-title">{`Resume ${titleNoun}`}</h1>
                     </div>
                     <div className="modal-body">
-                        {this.renderBody()}
+                        {this.renderBody(multi, instances)}
                     </div>
                     <div className="modal-footer">
                         <button type="button" className="btn btn-danger" onClick={this.cancel}>
                             Cancel
                         </button>
                         <button type="button" className="btn btn-primary" onClick={this.confirm}>
-                            Yes, resume this instance
+                            {`Yes, resume ${buttonText}`}
```

Comment:
Likewise, it seems that these modals seem to have "A lot in common" and that, largely, their difference lies in which `Instance Action` should be used. Perhaps there is a way that we can generalize this to avoid redundancy.